src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -2080,6 +2125,11 @@ private void WriteRedMinimalFactory(Node nd, bool withStringNames = false)
                         if (hasPreviousParameter)
                             Write(", ");
 
+                        if (IsNode(field.Type) && !IsOptional(field) && field.Type != "SyntaxToken")
```

Comment:
interesting that `?` is added when it's not optional.  can you comment the logic there?

Reply:
maybe IsNonOptionalReferenceType.

Reply:
> interesting that `?` is added when it's not optional. can you comment the logic there?

This is super obvious in commit d2c850300a0cb3b3accde0da9632f238dfd589d5 with the change. The field type is not nullable, but the type is being used in a location where it is nullable.